src/cpu.c

```diff
@@ -363,6 +368,37 @@ void OPC_JR_C(void) {
     OPC_JR_cc_n(get_flag(C_FLAG), 1);
 }
 
+static void OPC_JP_cc_a16(uint8_t bit, uint8_t branching_condition) {
+    uint16_t n = mmu_get_two_bytes(cpu.PC + 1);
+
+    // doing the PC increment before the opcode's override because we are emulating
+    // 3 CPU cycles beforehand (fetching opcode in cpu_step, and fetching n above)
+    cpu.PC += 3;
+
+    if (bit == branching_condition) {
+        cpu.PC = n;
+        cpu.cycle_count += 16;
+    } else {
+        cpu.cycle_count += 12;
+    }
```

Comment:
```suggestion
    if (bit == branching_condition) {
        cpu.PC = n;
        cpu.cycle_count += 16;
    } else {
        cpu.PC += 3;
        cpu.cycle_count += 12;
    }
```

I feel like the +3 should be in the else, since it is useless for the if branch.

Reply:
I would disagree, as the PC update reflects the three fetching instructions done to get to the current state

Reply:
But of what purpose is reflecting this fact? The instruction is "atomic" in a sense 🤔

Reply:
Sure but then the question remains, why are we even incrementing the PC if we don't know why we do this
This additionally keeps consistency with `JR cc, n`

Reply:
But we do know why we do this? In the else branch, we consume whatever is left of the instruction to update the PC. In the if branch we simply overwrite it. If we want to explain anything about it, I would argue that it should be done by making a comment. But using code this way doesn't seem appropriate to me to indicate "why" we are doing something.

I agree that they should be kept consistent but the same argument that I'm making here holds for `JR cc, n`

Reply:
At the end of the day, I can change it no problem
But in the JR the problem arises that we have duplicated code, as the updated PC is used in both branches

Reply:
That would be great, but if you deem it a good idea, we could add a comment to explain what's happening 🤔 

I thought that it's the same scenario:
![image](https://user-images.githubusercontent.com/33261455/199369381-6bd39fb8-1f08-4ff6-86f5-89e99e319ad2.png)

Reply:
Yeah but in the if-branch one can see that we add `cpu.PC + n` to `cpu.PC`, where `cpu.PC` is updated with `+=2`
Therefore, we don't discard the incrementation by two but use it in both branches

Reply:
Ouh shoot, you are right, I missed that 🤔 

But then it's fine to leave JR like that. I mean there is no consistency between JR and JP then. We would artificially create it 🤔